### ISSUE TITLE
BigQuery data load fails from GCS on the first load.

### ISSUE THREAD

**Issue by lukesneeringer** [GoogleCloudPlatform/google-cloud-python#3602]
This issue is re-posted from an internal ticket. (Googlers: b/63330731)

--------------------------------

Using the google-cloud-python BigQuery client, loading data from GCS fails repeatably on first load due to schema issues, then succeeds on retry.  Using _any_ other method of BQ load from GCS, this behavior is not observed.  Occurs on version 0.24 and 0.25 of the BigQuery client.

GCS source objects are AVRO files.

The error indicates an incorrect extra field in the schema:

```python
[{u'reason': u'invalid', u'message': u'Provided Schema does not match Table <redacted>. Cannot add fields (field: com_<redacted>_<redacted>_Content_value).'}]
```

As a workaround, the customer reimplemented the BQ load from GCS process using JSON api, which also works correctly.

**Comment by dhermes** [comment]
@lukesneeringer Any code snippet?

**Comment by tseaver** [comment]
#2926 might be relevant.

**Comment by lukesneeringer** [comment]
@dhermes Sorry, that was all they gave me. I can ask for a snippet, though.

**Comment by lukesneeringer** [comment]
I have a snippet; sending out-of-band for privacy reasons.

**Comment by lukesneeringer** [comment]
@dhermes @tseaver Did either of you have a chance to investigate this?

**Comment by dhermes** [comment]
@lukesneeringer I did not

**Comment by lukesneeringer** [comment]
Is that something you could do today or tomorrow?

**Comment by dhermes** [comment]
I'd prefer to punt to @tseaver

**Comment by chemelnucfin** [comment]
Would this be fixed by now?

**Comment by chemelnucfin** [comment]
@lukesneeringer Would you be able to hand me that snippet so I can take a look?

**Comment by chemelnucfin** [comment]
@lukesneeringer I believe that this has been fixed and confirmed via #4821.  Please feel free to reopen if it's not.  You may also be able to close the internal ticket also.  Thanks.

**Comment by DrValani** [comment]
Is this released in google-cloud-bigquery==0.31.0. If so then, I am still getting this error.
